File: app/src/main/java/com/timotiusoktorio/pencake/ui/dialogs/TimePickerFragment.java
package com.timotiusoktorio.pencake.ui.dialogs;

import android.app.Dialog;
import android.app.DialogFragment;
import android.app.TimePickerDialog;
import android.content.Context;
import android.os.Bundle;
import android.text.format.DateFormat;

import java.util.Calendar;

public class TimePickerFragment extends DialogFragment {

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (!(context instanceof TimePickerDialog.OnTimeSetListener)) {
            throw new IllegalStateException(context.getClass().getSimpleName() +
                    " must implements TimePickerDialog.OnTimeSetListener");
        }
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        final Calendar calendar = Calendar.getInstance();
        int hour = calendar.get(Calendar.HOUR_OF_DAY);
        int minute = calendar.get(Calendar.MINUTE);
        return new TimePickerDialog(getActivity(), (TimePickerDialog.OnTimeSetListener) getActivity(),
                hour, minute, DateFormat.is24HourFormat(getActivity()));
    }
}

File: app/src/main/java/com/timotiusoktorio/pencake/ui/dialogs/TimePickerFragment.kt
package com.timotiusoktorio.pencake.ui.dialogs

import android.app.Dialog
import android.app.DialogFragment
import android.app.TimePickerDialog
import android.content.Context
import android.os.Bundle
import android.text.format.DateFormat
import java.util.*

class TimePickerFragment : DialogFragment() {

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context !is TimePickerDialog.OnTimeSetListener) {
            throw IllegalStateException("${context.javaClass.simpleName} must implements TimePickerDialog.OnTimeSetListener")
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val calendar = Calendar.getInstance()
        val hour = calendar.get(Calendar.HOUR_OF_DAY)
        val minute = calendar.get(Calendar.MINUTE)
        return TimePickerDialog(activity, activity as TimePickerDialog.OnTimeSetListener,
                hour, minute, DateFormat.is24HourFormat(activity))
    }
}